native/packages/react-native-bpk-component-text-input/src/BpkTextInputIcons.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { StyleSheet, I18nManager } from 'react-native';
+import BpkIcon from 'react-native-bpk-component-icon';
+import { colorRed500, colorGreen500, spacingSm } from 'bpk-tokens/tokens/base.react.native';
+
+// Required so that it appears in the same place as the clear button.
+// TODO put these into tokens because this isn't so nice.
+const iconPlacement = I18nManager.isRTL ? 9 : spacingSm * 1.5;
```

Comment:
I feel the need to point out how awful this is. We have to hard set this so that the icon overlaps iOS's close icon just right. In RTL it's placed differently for some reason (they couldn't even make it an even number!).

As @k0nserv rightly points out, this is not a good long term solution as if iOS changes, it breaks our layout. We should soon look at rendering our icons outside of the text field to prevent this.